Cu 86c00w97j feature enhnace app navigation

basemosama opened the pull request:
**New Playx Navigation package.**
  - Add extra object to AppRouter navigation methods
  - Enhance app navigation onEnter and onExit
  - Update PlayxRoute by adding new PlayxPageTransition and PlayxPageConfiguration.
  
    -   Add PlayxPageTransition to determine route transition more easier.
    -   Add PlayxPageConfiguration to configure route page like creating custom transition.
    -   Removed pageBuilder and made builder to be required as we creating the page based on PlayxPageTransition and PlayxPageConfiguration.
    
**Use updated playx package with new playx navigation to enhance navigation.**
  -   Updated Playx Theme package
  -   Updated other packages.

basemosama committed:
feat: Add extra object to AppRouter navigation methods. -CU-86c00w97j

basemosama committed:
feat: Update PlayxBinding onExit function to include GoRouterState. -CU-86c00w97j

basemosama committed:
feat: Update PlayxRoute by adding new PlayxPageTransition and PlayxPageConfiguration  . -CU-86c00w97j

Add PlayxPageTransition to determine route transition more easier.
Add PlayxPageConfiguration to configure route page like creating custom transition.
Removed pageBuilder and made builder to be required as we creating the page based on PlayxPageTransition and PlayxPageConfiguration.

basemosama committed:
feat: Enhance app navigation onEnter and onExit. -CU-86c00w97j

basemosama committed:
refactor: Code cleanup. -CU-86c00w97j

basemosama committed:
refactor: Code cleanup. -CU-86c00w97j

mutasimissa commented:
Task linked: [CU-86c00w97j Feature, Enhnace app navigation](https://app.clickup.com/t/86c00w97j)

basemosama committed:
feat: Update playx to v0.1.2 . -CU-86c00w97j

Use updated playx package with new playx navigation to enhance navigation.
Updated Playx Theme package
Updated other packages.

mutasimissa commented:
@basemosama are you merging this?

basemosama commented:
> @basemosama are you merging this?

Yes, I will review it and merge it today

basemosama committed:
feat: add sentry navigator observer -CU-86c00w97j
